File: LeetCode/Easy/0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun postorderTraversal(root: TreeNode?): List<Int> {
        if (root == null) return emptyList()
        
        val list = mutableListOf<Int>();
        list.addAll(postorderTraversal(root.left))
        list.addAll(postorderTraversal(root.right))
        list.add(root.`val`)
        
        return list
    }
}

File: LeetCode/Easy/0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        if not root: return []
        
        answer = []
        if root.left:
            answer += self.postorderTraversal(root.left)
        
        if root.right:
            answer += self.postorderTraversal(root.right)
        
        answer.append(root.val)
        return answer
